modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/dto/v1_0/util/ProcessUtil.java

```diff
@@ -68,26 +66,28 @@ public static Process toProcess(Document document, Locale locale) {
 	}
 
 	private static Map<String, String> _createTitleMap(Document document) {
-		return Stream.of(
-			document.getFields()
-		).map(
-			Map::entrySet
-		).flatMap(
-			Collection::stream
-		).filter(
-			entry ->
-				StringUtil.startsWith(entry.getKey(), "title_") &&
-				!StringUtil.endsWith(entry.getKey(), "_sortable")
-		).collect(
-			Collectors.toMap(
-				entry -> _toLanguageTag(
-					StringUtil.removeSubstring(entry.getKey(), "title_")),
-				entry -> {
-					Field field = entry.getValue();
-
-					return String.valueOf(field.getValue());
-				})
-		);
+		Map<String, Field> fields = document.getFields();
+
+		HashMapBuilder.HashMapWrapper<String, String> hashMapWrapper =
+			HashMapBuilder.create(fields.size());
```

Comment:
@slnn Please don't use `HashMapBuilder` here. Just declare a `HashMap` named `titleMap` (matching the method name), and put key-values in the loop.

Additionally, we usually don't give `HashMap` or any other hash based data structure initial capacity, which is due to how hash data structures are implemented.

Please make changes to all the occasions in your PR and resend, thanks!